### PR TITLE
UF-320: Preferences central UI + Admin Page

### PR DESCRIPTION
**Pull request by paulovmr**

### CODE REVIEW

**Comment by paulovmr** [comment]
Merge together:

https://github.com/uberfire/uberfire/pull/530
https://github.com/droolsjbpm/kie-wb-common/pull/509
https://github.com/droolsjbpm/jbpm-form-modeler/pull/70
https://github.com/droolsjbpm/drools-wb/pull/318
https://github.com/droolsjbpm/jbpm-designer/pull/472
https://github.com/droolsjbpm/jbpm-console-ng/pull/506
https://github.com/droolsjbpm/optaplanner-wb/pull/112
https://github.com/droolsjbpm/livespark/pull/56
https://github.com/droolsjbpm/kie-wb-distributions/pull/378

**Comment by ederign** [comment]
jenkins retest this

**Comment by paulovmr** [comment]
jenkins retest this

File: uberfire-docs/docs/uberfireExtensions/adminPage.md [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+                             "Preferences" );
+}
+```
+
+As this example shows, you can add an admin tool to the page in three different ways:
+* Simple tool: Just adds a tool with an icon and a text. When the tool is selected (clicked), the command passed will be executed.
+* Tool with counter: It's like a simple tool, but it shows a counter under the text. The counter is computed using the passed parameterized command, when the Admin Page is opened.
+* Preference tool: It's similar to a simple tool, but instead of passing a command, you just have to inform the identifier of the preference that will have its edition screen opened when the tool is clicked.
+
+All tools above were inserted inside categories (e.g. Securty, General and Preferences). A category basically define how the tools are grouped. Each group is a line, possibly broke into more lines, and arranged based on the call order (first category inserted comes first).
```
Comment (csadilek) [inline uberfire-docs/docs/uberfireExtensions/adminPage.md]: possibly broken down into multiple lines...

File: uberfire-docs/docs/uberfireExtensions/preferencesEditionScreen.md [hunk 2]
```diff
@@ -0,0 +1,164 @@
+# Edition Screen
+
+## Introduction
+
+Uberfire provides a easy and customizable way for preferences edition through the screen.
```
Comment (csadilek) [inline uberfire-docs/docs/uberfireExtensions/preferencesEditionScreen.md]: maybe better...an easy and customizable way for editing preferences using screens.

File: uberfire-docs/docs/uberfireExtensions/preferencesEditionScreen.md [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+
+This will insert the admin tool "My Preference" to the Admin Page, and group it together with all tools of the "Preferences" group (in this example, only "Shared Preferences"):
+
+![adminPage](adminPage.png)
+
+When the user clicks on "My Preference", the following screen will be opened:
+
+![preferencesEditionScreen](preferencesEditionScreen.png)
+
+In the left, a navigation tree can observed. It is automatically built, based on the preference hierarchy defined by your beans.
```
Comment (csadilek) [inline uberfire-docs/docs/uberfireExtensions/preferencesEditionScreen.md]: A navigation tree is displayed on the left.

File: uberfire-docs/docs/uberfireExtensions/preferencesEditionScreen.md [hunk 4]
```diff
@@ -0,0 +1,164 @@
...
+
+In the right, a edition form can be observed. It can be automatically generated (you have to do nothing!), or it can be customized. When using the automatically generated form, you should change the attribute `formType` on `@Property` to choose the type of field that will be displayed on the form. The types of field supported are all in `MyPreference` example. By default, it will be a text field.
```
Comment (csadilek) [inline uberfire-docs/docs/uberfireExtensions/preferencesEditionScreen.md]: see above

**Comment by paulovmr** [comment]
@csadilek Thanks for the review! I've made the suggested changes.

**Comment by csadilek** [comment]
@paulovmr thanks, ready to go, imo. @ederign wanna take a look before I merge this all?

**Comment by ederign** [comment]
@csadilek go ahead and merge it! I've already follow paulo's work during development.

**Pull request merged**